amcl/src/amcl_node.cpp

```diff
@@ -1511,12 +1654,31 @@ AmclNode::handleInitialPoseMessage(const geometry_msgs::PoseWithCovarianceStampe
 void
 AmclNode::applyInitialPose()
 {
-  boost::recursive_mutex::scoped_lock cfl(configuration_mutex_);
+  std::lock_guard<std::recursive_mutex> cfl(configuration_mutex_);
   if( initial_pose_hyp_ != NULL && map_ != NULL ) {
     pf_init(pf_, initial_pose_hyp_->pf_pose_mean, initial_pose_hyp_->pf_pose_cov);
     pf_init_ = false;
+      ROS_DEBUG("\nInitial pose: %6.3f %6.3f %6.3f",
+               initial_pose_hyp_->pf_pose_mean.v[0],
+               initial_pose_hyp_->pf_pose_mean.v[1],
+               initial_pose_hyp_->pf_pose_mean.v[2]);
+
 
     delete initial_pose_hyp_;
     initial_pose_hyp_ = NULL;
   }
 }
+
+std::string
+AmclNode::stripLeadingSlash(const std::string& frame_name)
+{
+  if (frame_name.size() > 0)
+    if (frame_name[0] == '/')
```

Comment:
This looks like a useful function, maybe put a TODO so that we move it to a place available for all packages using frames